static/js/components/CourseListItemWithPopover.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+import React from 'react';
+import Popover from 'material-ui/Popover';
+import PopoverNullAnimation from '../util/popover_animation';
+
+
+export default class CourseListItemWithPopover extends React.Component {
+  props: {
+    id: number,
+    title: string,
+    description: string,
+    url: string,
+    enrollment_text: string,
+  }
+
+  state = {
+    isOpen: false,
+  }
+
+  handleClick = (event) => {
+    // This prevents ghost click.
+    event.preventDefault();
+
+    this.setState({
+      isOpen: true,
+      anchorEl: event.currentTarget,
+    });
+  }
+
+  handleRequestClose = () => {
+    this.setState({
+      isOpen: false,
+    });
+  }
+
+  render() {
+    const {
+      title,
+      description,
+      url,
+      enrollment_text: enrollmentText
+    } = this.props;
+    const {
+      isOpen,
+      anchorEl
+    } = this.state;
+
+    const titleEl = (url) => (
+      url ? <a href={url}>{title}</a> : title
+    );
+    const popoverLink = (url) => (
+      url ? <a className="edx-link" href={url}>View on edX</a> : null
+    );
```

Comment:
I'd declare these two functions outside of `render`, they don't have any serious runtime dependency which they couldn't take as an argument.